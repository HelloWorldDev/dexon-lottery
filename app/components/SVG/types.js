export type SVGProps = {
  src?: string,
  onClick: () => void,
  className?: string,
};
