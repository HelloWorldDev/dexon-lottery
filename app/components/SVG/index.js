// @flow
import React from 'react';
import styled from 'styled-components';
import InlineSVG from 'react-inlinesvg';
import type { SVGProps } from './types';

export const StyledSVG = styled(InlineSVG)`
  display: flex;
  align-items: center;
`;

const SVG = ({ src, ...props }: SVGProps) => (src ? (
  <StyledSVG cacheGetRequests src={src} {...props} />
) : null);

export default SVG;
