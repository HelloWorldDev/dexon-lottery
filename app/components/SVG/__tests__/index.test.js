import React from 'react';
import { shallow } from 'enzyme';

import SVG, { StyledSVG } from '../';

jest.mock('react-inlinesvg', () => 'div');

describe('<SVG />', () => {
  it('should render empty component', () => {
    const wrapper = shallow(
      <SVG />
    );

    expect(wrapper.html()).toBeNull();
  });

  it('should render SVG', () => {
    const wrapper = shallow(
      <SVG src="/test.svg" />
    );

    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should pass props down to svg', () => {
    const wrapper = shallow(
      <SVG src="/test.svg" className="className" />
    );

    expect(wrapper.find(StyledSVG).prop('className')).toBe('className');
  });
});
