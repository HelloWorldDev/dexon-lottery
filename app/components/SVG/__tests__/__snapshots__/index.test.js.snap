// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SVG /> should render SVG 1`] = `"<styled.div cacheGetRequests={true} src=\\"/test.svg\\" />"`;
